### ISSUE TITLE
Use hostname as node name

### ISSUE THREAD

**Issue by JorritSalverda** [pires/docker-elasticsearch-kubernetes#32]
I run elastic in a statefulset where the hostnames are predictable. It would be nice if the name of the node is the same as the hostname in that case. Could this be configured or are there reasons not to do so?

**Comment by pires** [comment]
I don't think this is possible with Deployments.

**Comment by hikhvar** [comment]
We could at least set the elasticsearch node name via an environment variable. The Downward API allows us to get the Pod name as environment variable: 
```
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
```
I will prepare a pull request for this repo and the kubernetes configs since I think it is good to match the output of 
```
_cat/nodes?v
```
or other Elasticsearch node related information with the actual pod.

**Comment by hikhvar** [comment]
Added the pull request here: https://github.com/pires/docker-elasticsearch since, I needed to change the `run.sh`.

**Comment by hikhvar** [comment]
Here is the pull request to apply it to the kubernetes configs: https://github.com/pires/kubernetes-elasticsearch-cluster/pull/82

**Comment by pires** [comment]
Thanks @hikhvar. I'll close this and we'll pick up from the PR which will be merged today, with version 5.2.2.
